sys/net/credman/credman.c

```diff
@@ -66,24 +68,22 @@ int credman_add(const credman_credential_t *credential)
         ret = CREDMAN_TYPE_UNKNOWN;
         goto end;
     }
-    pos = _find_credential_pos(credential->tag, credential->type);
+
+    pos = _find_credential_pos(credential->tag, credential->type, &entry);
     if (pos >= 0) {
         DEBUG("credman: credential with tag %d and type %d already exist\n",
               credential->tag, credential->type);
         ret = CREDMAN_EXIST;
-        goto end;
     }
-    /* find the next free position in credential pool */
-    pos = _find_credential_pos(CREDMAN_TAG_EMPTY, CREDMAN_TYPE_EMPTY);
-    if (pos < 0) {
+    else if (entry == NULL) {
         DEBUG("credman: no space for new credential\n");
         ret = CREDMAN_NO_SPACE;
-        goto end;
     }
-
-    memcpy(&credentials[pos], credential, sizeof(*credential));
-    used++;
-    ret = CREDMAN_OK;
+    else {
+        memcpy(entry, credential, sizeof(credman_credential_t));
```

Comment:
Side-note: `*entry = *credential` does the same (and should have the same ROM-size) but is type-safe, so I would prefer it like this. But this is not blocking.